### PR TITLE
Adding summaries to the download metrics

### PR DESCRIPTION
**Pull request by AshCripps**
This pr adds `summaries.js` which takes an entires days (the day before) processed logs and produced a summation of all the downloads for that day.

<details>
<summary>Example output hidden in details:</summary>
<br>

```

{
  bytes: 20874017783579,
  total: 2409726,
  country: {
    US: 1632659,
    AU: 22700,
    GB: 32812,
    IE: 93722,
    DE: 130847,
    PT: 1670,
    NL: 98602,
    TN: 1134,
    BG: 948,
    MX: 2861,
    LV: 933,
    FR: 37339,
    BR: 16543,
    JP: 19280,
    NZ: 1076,
    TH: 2278,
    IN: 33332,
    NO: 1738,
    CL: 1034,
    CA: 18474,
    CN: 86016,
    SE: 5288,
    AR: 2281,
    ES: 5674,
    KR: 8533,
    TR: 3783,
    TW: 3409,
    PL: 8560,
    FI: 6016,
    SG: 12137,
    CH: 7329,
    BE: 10345,
    RU: 9037,
    IL: 3647,
    ID: 8475,
    AT: 4677,
    ZA: 1714,
    DO: 207,
    KE: 648,
    XX: 797,
    CO: 1976,
    SK: 704,
    EC: 504,
    MA: 994,
    HK: 6708,
    CZ: 3314,
    CR: 332,
    PK: 4277,
    VN: 5807,
    IT: 3930,
    NI: 58,
    BY: 1330,
    DK: 1918,
    BO: 294,
    PH: 2236,
    CY: 277,
    HR: 780,
    VE: 661,
    MK: 205,
    SA: 828,
    LK: 527,
    NG: 1457,
    AO: 313,
    BD: 2485,
    IQ: 230,
    PA: 104,
    UA: 4502,
    MR: 43,
    RO: 1726,
    GP: 8,
    PE: 765,
    PY: 89,
    RS: 701,
    BH: 103,
    MY: 1331,
    EG: 1687,
    YE: 244,
    UY: 300,
    EE: 519,
    BB: 21,
    SV: 161,
    PR: 73,
    SN: 119,
    GR: 644,
    GT: 179,
    JO: 189,
    CM: 391,
    KZ: 548,
    LT: 594,
    AE: 559,
    T1: 36,
    AM: 292,
    TT: 23,
    DZ: 710,
    HU: 1320,
    GH: 403,
    LB: 228,
    GM: 22,
    PF: 13,
    BA: 201,
    JM: 85,
    QA: 189,
    SI: 285,
    ML: 65,
    ZM: 44,
    TZ: 78,
    NP: 438,
    UZ: 383,
    UG: 343,
    IR: 2465,
    KH: 249,
    IS: 108,
    KN: 1,
    BN: 11,
    DJ: 3,
    LU: 330,
    HN: 74,
    CU: 453,
    AL: 144,
    OM: 33,
    FJ: 1,
    BQ: 1,
    MN: 152,
    SD: 63,
    KG: 117,
    BZ: 7,
    PS: 207,
    AZ: 123,
    PG: 13,
    SY: 144,
    LA: 521,
    MD: 171,
    CI: 142,
    GE: 155,
    ET: 200,
    AF: 168,
    BI: 42,
    MO: 19,
    MM: 111,
    KW: 123,
    CV: 21,
    BF: 41,
    MP: 1,
    CD: 60,
    ER: 22,
    NC: 2,
    SC: 24,
    MU: 74,
    VC: 2,
    RE: 75,
    MG: 405,
    MW: 53,
    RW: 60,
    SO: 9,
    MV: 8,
    TJ: 91,
    MT: 67,
    NA: 18,
    HT: 13,
    VG: 80,
    SZ: 2,
    SL: 3,
    MZ: 60,
    ZW: 190,
    BJ: 143,
    TL: 9,
    BW: 56,
    ME: 18,
    GA: 2,
    MC: 4,
    TG: 103,
    KM: 3,
    IM: 26,
    AG: 4,
    GI: 7,
    XK: 14,
    FO: 7,
    AD: 19,
    LS: 2,
    LI: 10,
    VU: 17,
    SM: 3,
    LY: 13,
    GG: 2,
    JE: 5,
    GU: 3,
    CW: 3,
    LR: 7,
    TD: 2,
    AX: 4,
    SR: 2,
    GF: 2,
    TM: 2,
    NE: 6,
    MQ: 6,
    LC: 4,
    TC: 3,
    CG: 3,
    BS: 3,
    GQ: 1,
    VI: 3,
    GN: 2,
    BT: 1,
    BM: 1,
    ST: 2,
    GY: 8,
    GL: 1
  },
  version: {
    'v12.18.0': 90970,
    'v12.13.0': 71196,
    'v6.17.1': 6437,
    'v8.9.0': 2256,
    'v10.15.0': 8914,
    'v10.23.1': 19941,
    'v14.15.4': 88164,
    'v12.18.3': 75311,
    'v10.23.3': 98020,
    'v12.20.2': 156156,
    'v12.16.2': 39050,
    'v0.10.35': 2594,
    'v14.15.5': 377515,
    'v10.15.3': 36927,
    'v11.15.0': 11638,
    'v12.18.4': 39730,
    'v12.14.1': 31541,
    'v13.0.0': 304,
    'v4.2.1': 57,
    'v12.19.1': 17374,
    'v14.15.3': 12780,
    'v12.20.0': 16451,
    'v8.10.0': 5890,
    'v12.20.1': 76604,
    'v15.7.0': 2833,
    'v12.5.0': 1216,
    'v12.19.0': 47478,
    'v8.11.3': 8892,
    'v10.22.0': 13414,
    'v14.15.1': 36924,
    'v8.17.0': 32650,
    'v13.14.0': 19749,
    'v9.11.1': 1162,
    'v6.11.1': 5818,
    'v13.9.0': 1271,
    'v10.16.3': 28112,
    'v10.19.0': 30616,
    'v14.7.0': 5204,
    'v10.14.2': 3749,
    'v12.7.0': 8874,
    'v14.5.0': 5863,
    'v14.0.0': 3130,
    'v4.3.2': 116,
    'v15.9.0': 130720,
    'v12.16.1': 59229,
    'v4.8.7': 140,
    'v10.17.0': 17337,
    'v8.16.2': 1952,
    'v15.5.1': 4567,
    'v15.0.0': 387,
    'v12.13.1': 24410,
    'v8.9.3': 4204,
    'v10.23.2': 13568,
    'v11.10.1': 1097,
    'v6.14.4': 1738,
    'v14.15.0': 33940,
    'v12.16.3': 30963,
    'v4.6.0': 315,
    'v8.7.0': 749,
    'v11.4.0': 456,
    'v10.18.0': 5594,
    'v8.11.1': 3218,
    'v10.13.0': 11610,
    'v14.2.0': 2380,
    'v14.12.0': 1658,
    'v10.14.0': 1267,
    'v12.3.1': 3047,
    'v12.18.1': 12361,
    'v6.13.1': 1180,
    'v12.18.2': 20485,
    'v12.15.0': 10302,
    'v15.6.0': 4461,
    'v6.16.0': 1034,
    'v12.14.0': 11083,
    'v8.11.4': 4601,
    'v13.0.1': 710,
    'v12.12.0': 3565,
    'v8.12.0': 9111,
    'v8.15.1': 1884,
    'v10.16.0': 27786,
    'v15.8.0': 8042,
    'v14.11.0': 3075,
    'v8.14.0': 1169,
    'v12.9.0': 764,
    'v10.23.0': 23316,
    unknown: 1217,
    'v10.16.2': 3488,
    'v14.3.0': 3545,
    'v9.3.0': 233,
    'v6.11.2': 1046,
    'v13.10.1': 1471,
    'v5.11.0': 23,
    'v11.12.0': 834,
    'v10.18.1': 10777,
    'v10.15.1': 7341,
    'v10.21.0': 11443,
    'v4.2.4': 190,
    'v10.8.0': 657,
    'v10.11.0': 1881,
    'v8.9.4': 7235,
    'v12.8.0': 1438,
    'v10.14.1': 6073,
    'v14.8.0': 5341,
    'v14.15.2': 6591,
    'v14.9.0': 3126,
    'v6.11.4': 255,
    'v9.11.2': 3858,
    'v10.10.0': 2245,
    'v13.6.0': 901,
    'v14.14.0': 4635,
    'v12.10.0': 7514,
    'v8.9.2': 330,
    'v10.22.1': 9757,
    'v6.11.3': 589,
    'v6.10.2': 1829,
    'v8.13.0': 430,
    'v10.9.0': 4704,
    'v12.8.1': 2488,
    'v6.9.2': 1343,
    'v12.16.0': 68874,
    'v10.15.2': 2113,
    'v4.4.7': 240,
    'v14.10.0': 1080,
    'v9.7.0': 78,
    'v15.3.0': 2501,
    'v6.0.0': 317,
    'v10.6.0': 711,
    'v6.2.2': 589,
    'v10.5.0': 1095,
    'v0.10.46': 480,
    'v11.7.0': 572,
    'v11.11.0': 836,
    'v4.9.1': 1611,
    'v10.20.0': 1138,
    'v12.0.0': 5401,
    'v8.4.0': 1405,
    'v10.20.1': 5049,
    'v0.10.36': 46,
    'v8.1.0': 160,
    'v13.12.0': 3164,
    'v7.10.1': 2754,
    'v10.12.0': 2417,
    'v8.2.0': 270,
    'v8.16.1': 1981,
    'v6.15.1': 534,
    'v14.4.0': 6385,
    'v12.17.0': 15429,
    'v15.0.1': 1829,
    'v9.6.1': 204,
    'v6.9.1': 1376,
    'v8.11.0': 895,
    'v5.12.0': 568,
    'v4.1.1': 1282,
    'v8.1.4': 944,
    'v13.7.0': 1695,
    'v12.6.0': 8275,
    'v8.16.0': 3735,
    'v15.4.0': 3206,
    'v8.5.0': 479,
    'v6.13.0': 34,
    'v12.2.0': 1490,
    'v7.2.1': 193,
    'v6.10.3': 1884,
    'v4.8.6': 73,
    'v13.8.0': 2715,
    'v6.14.3': 275,
    'v12.11.0': 519,
    'v9.4.0': 468,
    'v0.8.10': 8,
    'v0.10.31': 541,
    'v8.15.0': 2221,
    'v6.9.5': 843,
    'v14.13.1': 4514,
    'v11.14.0': 1588,
    'v6.7.0': 513,
    'v11.5.0': 243,
    'v6.14.2': 99,
    'v14.6.0': 2995,
    'v8.14.1': 514,
    'v6.9.4': 616,
    'v6.11.5': 403,
    'v0.10.48': 1515,
    'v4.0.0': 396,
    'v0.12.18': 927,
    'v8.9.1': 2606,
    'v10.4.1': 1678,
    'v11.3.0': 638,
    'v13.2.0': 450,
    'v8.6.0': 1574,
    'v15.5.0': 1405,
    'v8.2.1': 368,
    'v15.2.0': 854,
    'v12.9.1': 2849,
    'v5.4.0': 53,
    'v6.1.0': 607,
    'v6.11.0': 442,
    'v13.13.0': 1463,
    'v10.3.0': 384,
    'v13.11.0': 2078,
    'v6.6.0': 150,
    'v11.9.0': 695,
    'v5.7.1': 29,
    'v6.8.0': 48,
    'v12.3.0': 291,
    'v8.11.2': 2681,
    'v4.2.6': 299,
    'v9.8.0': 489,
    'v6.17.0': 185,
    'v6.10.0': 513,
    'v12.1.0': 2357,
    'v12.4.0': 4138,
    'v4.5.0': 706,
    'v14.10.1': 1218,
    'v7.10.0': 389,
    'v4.2.2': 87,
    'v14.13.0': 4097,
    'v4.4.5': 436,
    'v6.2.1': 167,
    'v7.4.0': 417,
    'v8.0.0': 445,
    'v9.0.0': 388,
    'v9.9.0': 297,
    'v0.10.44': 48,
    'v11.10.0': 742,
    'v10.7.0': 1502,
    'v0.8.18': 3,
    'v13.5.0': 2843,
    'v6.9.0': 523,
    'v12.11.1': 1871,
    'v11.6.0': 627,
    'v4.4.3': 157,
    'v8.1.2': 294,
    'v7.7.0': 256,
    'v8.8.1': 514,
    'v0.6.8': 3,
    'v10.2.0': 109,
    'v9.11.0': 238,
    'v6.12.2': 152,
    'v5.2.0': 125,
    'v10.0.0': 1111,
    'v14.1.0': 1201,
    'v10.2.1': 421,
    'v10.1.0': 924,
    'v15.1.0': 860,
    'v9.2.1': 215,
    'v9.5.0': 456,
    'v6.12.3': 157,
    'v0.12.7': 407,
    'v4.6.1': 455,
    'v7.7.2': 49,
    'v15.2.1': 2099,
    'v11.0.0': 713,
    'v8.3.0': 567,
    'v10.16.1': 1371,
    'v4.1.0': 12,
    'v0.8.24': 16,
    'v5.10.1': 186,
    'v0.11.6': 3,
    'v9.10.1': 219,
    'v0.12.2': 632,
    'v0.10.26': 333,
    'v0.10.9': 7,
    'v5.4.1': 109,
    'v5.3.0': 74,
    'v7.9.0': 243,
    'v13.1.0': 375,
    'v0.6.11': 2,
    'v6.5.0': 131,
    'v6.3.1': 778,
    'v11.1.0': 420,
    'v0.10.39': 52,
    'v7.5.0': 127,
    'v11.8.0': 330,
    'v4.2.3': 181,
    'v13.3.0': 1209,
    'v0.8.28': 131,
    'v0.12.3': 2,
    'v13.10.0': 438,
    'v0.12.1': 14,
    'v6.14.0': 143,
    'v9.2.0': 278,
    'v6.12.1': 253,
    'v4.9.0': 17,
    'v7.8.0': 271,
    'v6.9.3': 68,
    'v4.8.2': 100,
    'v4.7.1': 8,
    'v4.3.1': 56,
    'v4.1.2': 110,
    'v0.8.6': 2,
    'v0.12.0': 137,
    'v4.3.0': 27,
    'v6.4.0': 120,
    'v10.4.0': 864,
    'v0.10.41': 56,
    'v6.3.0': 94,
    'v0.10.24': 49,
    'v0.10.16': 57,
    'v0.11.0': 26,
    'v9.10.0': 218,
    'v0.10.38': 65,
    'v0.10.32': 99,
    'v0.10.11': 34,
    'v11.13.0': 625,
    'v4.4.4': 80,
    'v6.14.1': 152,
    'v9.1.0': 105,
    'v0.12.14': 24,
    'v0.1.101': 1,
    'v7.7.4': 77,
    'v0.12.9': 71,
    'v6.12.0': 92,
    'v11.2.0': 300,
    'v4.7.3': 36,
    'v7.2.0': 65,
    'v0.10.13': 43,
    'v0.8.20': 37,
    'v0.10.28': 32,
    'v0.10.30': 46,
    'v0.6.20': 68,
    'v0.8.19': 33,
    'v0.8.23': 34,
    'v7.6.0': 133,
    'v0.8.2': 35,
    'v0.6.17': 35,
    'v0.10.5': 35,
    'v4.2.0': 14,
    'v6.2.0': 68,
    'v0.10.29': 144,
    'v6.8.1': 208,
    'v0.12.4': 68,
    'v5.1.0': 35,
    'v0.12.5': 15,
    'v7.1.0': 186,
    'v7.7.3': 76,
    'v0.10.18': 13,
    'v5.9.1': 82,
    'v5.6.0': 102,
    'v5.1.1': 60,
    'v0.11.16': 64,
    'v7.0.0': 178,
    'v0.10.33': 99,
    'v0.12.16': 74,
    'v6.10.1': 350,
    'v9.7.1': 436,
    'v0.10.17': 6,
    'v8.8.0': 93,
    'v5.7.0': 35,
    'v5.9.0': 21,
    'v5.10.0': 16,
    'v7.7.1': 81,
    'v0.10.1': 11,
    'v0.1.25': 1,
    'v0.12.6': 42,
    'v0.9.0': 3,
    'v0.10.0': 10,
    'v0.11.5': 4,
    'v0.10.15': 9,
    'v6.15.0': 108,
    'v8.1.3': 191,
    'v4.8.1': 9,
    'v5.0.0': 112,
    'v0.10.45': 11,
    'v5.5.0': 161,
    'v0.8.0': 5,
    'v0.3.4': 1,
    'v0.1.26': 1,
    'v0.10.23': 7,
    'v0.1.32': 1,
    'v9.6.0': 53,
    'v0.10.43': 8,
    'v0.6.13': 1,
    'v0.10.40': 71,
    'v0.7.0': 1,
    'v0.7.11': 1,
    'v0.10.21': 8,
    'v4.8.0': 47,
    'v0.10.42': 13,
    'v0.9.3': 6,
    'v4.8.4': 99,
    'v0.8.1': 2,
    'v4.4.6': 45,
    'v4.4.2': 16,
    'v0.9.2': 4,
    'v0.10.3': 11,
    'v0.11.4': 3,
    'v0.9.1': 6,
    'v0.9.9': 3,
    'v7.3.0': 82,
    'v0.6.7': 1,
    'v4.6.2': 76,
    'v0.10.25': 42,
    'v4.4.1': 19,
    'v4.8.3': 51,
    'v0.6.21': 18,
    'v0.6.15': 1,
    'v0.8.11': 3,
    'v0.10.47': 3,
    'v0.9.5': 4,
    'v0.12.12': 18,
    'v0.1.30': 1,
    'v0.10.4': 6,
    'v4.4.0': 44,
    'v13.4.0': 306,
    'v0.11.9': 6,
    'v0.8.22': 29,
    'v0.12.17': 14,
    'v0.10.8': 4,
    'v0.10.2': 5,
    'v0.12.10': 21,
    'v4.2.5': 6,
    'v0.6.19': 1,
    'v5.8.0': 23,
    'v0.6.14': 1,
    'v0.11.1': 6,
    'v0.9.7': 10,
    'v5.11.1': 33,
    'v8.1.1': 73,
    'v0.8.15': 9,
    'v0.6.5': 2,
    'v0.7.12': 4,
    'v0.9.12': 25,
    'v0.6.9': 2,
    'v0.12.13': 4,
    'v0.7.6': 2,
    'v0.7.3': 1,
    'v0.8.17': 6,
    'v0.1.100': 3,
    'v0.10.12': 4,
    'v0.10.20': 5,
    'v0.8.16': 4,
    'v0.9.10': 4,
    'v0.11.14': 9,
    'v0.12.8': 10,
    'v4.7.0': 28,
    'v0.4.12': 2,
    'v0.7.7': 1,
    'v0.6.6': 4,
    'v4.7.2': 50,
    'v0.7.1': 1,
    'v0.12.15': 14,
    'v0.11.7': 2,
    'v0.10.6': 2,
    'v0.11.10': 2,
    'v0.8.9': 3,
    'v0.11.11': 4,
    'v0.11.13': 3,
    'v4.8.5': 17,
    'v0.7.10': 1,
    'v0.1.20': 2,
    'v0.1.15': 2,
    'v0.8.14': 3,
    'v0.10.34': 3,
    'v0.8.8': 3,
    'v0.8.27': 11,
    'v0.6.12': 4,
    'v0.3.7': 1,
    'v0.9.11': 2,
    'v0.6.4': 1,
    'v0.6.3': 8,
    'v0.8.4': 1,
    'v0.8.26': 2,
    'v0.8.7': 7,
    'v0.2.0': 1,
    'v0.8.25': 2,
    'v0.1.103': 2,
    'v0.10.37': 48,
    'v0.10.22': 2,
    'v0.10.19': 3,
    'v0.8.21': 3,
    'v0.7.4': 1,
    'v0.10.27': 3,
    'v0.10.14': 3,
    'v0.1.14': 1,
    'v0.11.12': 2,
    'v0.10.10': 2,
    'v0.8.12': 1,
    'v0.10.7': 21,
    'v0.11.2': 1,
    'v0.11.8': 1,
    'v0.9.4': 1,
    'v0.8.13': 1,
    'v0.11.15': 1,
    'v0.9.8': 1,
    'v0.12.11': 1,
    'v0.11.3': 1,
    'v0.9.6': 1,
    'v0.8.5': 9,
    'v0.6.10': 1
  },
  os: {
    headers: 1196049,
    win: 237273,
    linux: 865904,
    src: 50290,
    osx: 58618,
    aix: 614,
    sunos: 978
  },
  arch: {
    unknown: 1246345,
    x64: 1139052,
    x86: 14246,
    armv7l: 2332,
    arm64: 4048,
    s390x: 1202,
    ppc64: 759,
    ppc64le: 1167,
    armv6l: 575
  }
}

```

</details>

I need to add back the feature of sorting the data but am opening this for people to check it over before I move on to the next step

### CODE REVIEW

**Comment by AshCripps** [comment]
ive pushed up a change that outputs the data into a file format using the same naming convention as current on the webserver for our metrics - however this output is in JSON and the old format is in CSV so im not sure how I can splice the two together, also there are a lot more files under https://nodejs.org/metrics/summaries/. Thoughts @rvagg 

also @nodejs/build this could do with some general reviews

**Comment by AshCripps** [comment]
The latest commit moves `summaries.js` to its own folder and dockerises it to run it in cloud function GCP

File: ansible/roles/metrics/files/summaries/summaries.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+      const schunk = chunk.toString()
+      if (!schunk.startsWith('day,country')) { // ignore header
+        line.push(schunk.split(','))
+        return(line)
+      }
+      return
+    } catch (e) {
+      console.log(e)
+    }
+
```
Comment (mhdawson) [inline ansible/roles/metrics/files/summaries/summaries.js:29]: Indentation/extra blank space

File: ansible/roles/metrics/files/summaries/summaries.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+      console.log('Upload complete')
+    }
+  })
+}
+
+
+app.post('/', async (req, res) => {
+
+  await produceSummaries()
+  res.status(200).send();
```
Comment (mhdawson) [inline ansible/roles/metrics/files/summaries/summaries.js:123]: Most places don't have the semicolon.  I prefer having it and the core Node.js library code uses semicolons,  but I think we should at least be consistent across the file.

File: ansible/roles/metrics/files/summaries/summaries.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  increment('version', version)
+  increment('os', os)
+  increment('arch', arch)
+  counts.bytes += parseInt(bytes, 10)
+  counts.total++
+  return
+ }
+
+async function collectData () {
+
```
Comment (mhdawson) [inline ansible/roles/metrics/files/summaries/summaries.js:74]: extra space

File: ansible/roles/metrics/files/summaries/summaries.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    const stringContents = data[0].toString()
+    const contentsArray = stringContents.split('\n')
+          
+    for (const line of contentsArray) {
+      try {
+        const csvparse = csvStream(line)
+        if (csvparse !== undefined && csvparse[0][0] !== '') { summary(csvparse) }
+      } catch (err) { console.log(err) }
+    }
+    }
```
Comment (mhdawson) [inline ansible/roles/metrics/files/summaries/summaries.js:96]: indentation

File: ansible/roles/metrics/files/summaries/summaries.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+      const line = []
+      const schunk = chunk.toString()
+      if (!schunk.startsWith('day,country')) { // ignore header
+        line.push(schunk.split(','))
+        return(line)
+      }
+      return
+    } catch (e) {
+      console.log(e)
+    }
```
Comment (mhdawson) [inline ansible/roles/metrics/files/summaries/summaries.js:27]: Extra indent.

**Review by mhdawson — APPROVED** [review]
LGTM

**Pull request merged**